### PR TITLE
identity token example for Azure

### PR DESCRIPTION
**Pull request by zainq11**
This PR aims to add an example for creating the necessary Dynamic credential setup for Stacks operation on an Azure account.

### CODE REVIEW

File: aws/main.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ resource "aws_iam_openid_connect_provider" "stacks_openid_provider" {
 
 resource "aws_iam_role" "stacks_role" {
   name               = "stacks-${var.tfc_organization}-${var.tfc_project}-${var.tfc_stack}"
-  assume_role_policy = data.aws_iam_policty_document.stacks_role_policy.json
+  assume_role_policy = data.aws_iam_policy_document.stacks_role_policy.json
```
Comment (zainq11) [inline aws/main.tf:16]: Fixed a typo here. Can be ignored if N/A.

Comment (liamcervante) [reply aws/main.tf:16]: Nice, I fixed this here already: https://github.com/liamcervante/terraform-stacks-identity-tokens/commit/1b24d1c3512b1a23d1b5e6732830837c8994a852. It's not identifying a conflict though, so we can just merge this as well 😅

File: azure/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+This directory contains the terraform code to create an Azure Service Principal and the Federated Identity credentials which will be used in later Terraform Stacks operations.
+
+## Setup
+
+To execute this configuration you will need to give Terraform access to the relevant Azure account. Refer the [following](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#authenticating-to-azure) terraform provider docs to authenticate with Azure.
+
+Once the Terraform configuration has been applied you should save the following from your Azure account:
+
+1. `Tenant ID` of the Azure AD/Microsoft Entra ID service.
+2. `Application (client) ID` of the registered application created by this terraform setup.
```
Comment (liamcervante) [inline azure/README.md]: Could we add this as an output to the configuration, just so it prints it out?

**Review by liamcervante — APPROVED** [review]
Looks good, just one thought around printing outputs from the configuration - but if you can't be bothered with it that's fine! Let me know and I can press the button with or without!

File: azure/README.md (hunk 2 above)
Comment (zainq11) [reply azure/README.md]: Good idea. I will make this change.

Comment (zainq11) [reply azure/README.md]: Added the outputs. Here is an example of how it looks
```bash
Outputs:

client_id = "c22094b1-047d-4221-bc61-82907b50f338"
subscription_id = "1845c88e-c6c2-4dfb-94e1-552aeecb228e"
tenant_id = "8e688f32-c820-4e21-97c7-49b5adee5220"
```

**Comment by liamcervante** [comment]
Thanks @zainq11!

**Pull request merged**